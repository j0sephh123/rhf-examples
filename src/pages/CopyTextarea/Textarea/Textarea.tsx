import { SetValue } from "../types";

type Props = {
  value: string;
  onChange: SetValue;
};

export default function Textarea({ value, onChange }: Props) {
  return (
    <>
      <textarea
        autoFocus
        className="textarea textarea-lg w-full max-w-5xl border-zinc-400"
        rows={6}
        value={value}
        onChange={(e) => onChange(e.target.value)}
      ></textarea>
    </>
  );
}
