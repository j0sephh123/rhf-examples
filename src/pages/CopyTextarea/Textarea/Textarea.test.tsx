import { render, screen, fireEvent } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import Textarea from "./Textarea";

describe("Textarea", () => {
  it("set its value to the value prop, on typing calls onChange callback with the correct value", async () => {
    const spy = vi.fn();
    const { asFragment } = render(<Textarea value="1" onChange={spy} />);
    const nativeTextarea = screen.getByRole("textbox");

    expect(nativeTextarea).toHaveValue("1");

    fireEvent.change(nativeTextarea, {
      target: { value: "2" },
    });

    expect(spy).toHaveBeenCalledWith("2");
    expect(asFragment()).toMatchSnapshot();
  });
});
