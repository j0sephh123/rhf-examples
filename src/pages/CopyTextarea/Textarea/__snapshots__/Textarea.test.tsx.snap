// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Textarea > set its value to the value prop, on typing calls onChange callback with the correct value 1`] = `
<DocumentFragment>
  <textarea
    class="textarea textarea-lg w-full max-w-5xl border-zinc-400"
    rows="6"
  >
    1
  </textarea>
</DocumentFragment>
`;
